interface/repository/csv_pokemon_repository.go

```diff
@@ -0,0 +1,148 @@
+package repository
+
+import (
+	"io"
+	"strconv"
+
+	"github.com/FernandoGal25/academy-go-q42021/domain/model"
+	customErrors "github.com/FernandoGal25/academy-go-q42021/error"
+	"github.com/FernandoGal25/academy-go-q42021/infrastructure/datastore"
+)
+
+/*
+	Repository that handles the operations of pokemon in a CSV file.
+*/
+type CSVPokemonRepository struct {
+	Handler *datastore.CSVHandler
+}
+
+/*
+	Returns an instance of CSVPokemonRepository.
+*/
+func NewCSVPokemonRepository(h *datastore.CSVHandler) CSVPokemonRepository {
+	return CSVPokemonRepository{h}
+}
+
+func buildPokemon(p *model.Pokemon, data []string, ID uint64) error {
+	p.ID = ID
+
+	p.Name = data[1]
+
+	h, err := strconv.Atoi(data[2])
+	if err != nil {
+		return customErrors.CSVFormatError{Message: "CSV ITEM DOES NOT POSSESS A VALID HEIGHT", Err: err}
+	}
+	p.Height = h
+
+	w, err := strconv.Atoi(data[3])
+	if err != nil {
+		return customErrors.CSVFormatError{Message: "CSV ITEM DOES NOT POSSESS A VALID WEIGHT", Err: err}
+	}
+	p.Weight = w
+
+	o, err := strconv.Atoi(data[4])
+	if err != nil {
+		return customErrors.CSVFormatError{Message: "CSV ITEM DOES NOT POSSESS A VALID ORDER", Err: err}
+	}
+	p.Order = o
+
+	b, err := strconv.Atoi(data[5])
+	if err != nil {
+		return customErrors.CSVFormatError{Message: "CSV ITEM DOES NOT POSSESS A VALID BASE_EXPERIENCE", Err: err}
+	}
+	p.BaseExperience = b
+
+	return nil
+}
+
+func unBuildPokemon(p *model.Pokemon) []string {
+	return []string{
+		strconv.FormatUint(p.ID, 10),
+		p.Name,
+		strconv.Itoa(p.Height),
+		strconv.Itoa(p.Weight),
+		strconv.Itoa(p.Order),
+		strconv.Itoa(p.BaseExperience),
+	}
+}
+
+/*
+	Searches the pokemon which ID belongs to.
+*/
+func (r CSVPokemonRepository) FindByID(ID uint64) (*model.Pokemon, error) {
+	if err := r.Handler.BuildHandler(); err != nil {
+		return nil, err
+	}
+	defer r.Handler.Close()
+	var i model.Pokemon
+
+	for {
+		record, err := r.Handler.Read()
+		if err == io.EOF {
```

Comment:
Better use `error.Is() method` to check for specific errors. In wrapped errors, comparing with `==` will result in giving a false negative.